packages/flutter_tools/test/general.shard/features_test.dart

```diff
@@ -71,358 +47,417 @@ void main() {
       expect(feature.getSettingForChannel('unknown'), masterSetting);
     });
 
-    testWithoutContext('env variables are only enabled with "true" string', () {
-      platform.environment = <String, String>{'FLUTTER_WEB': 'hello'};
-
-      expect(featureFlags.isWebEnabled, false);
-
-      platform.environment = <String, String>{'FLUTTER_WEB': 'true'};
-
-      expect(featureFlags.isWebEnabled, true);
-    });
-
-    testWithoutContext('Flutter web help string', () {
-      expect(flutterWebFeature.generateHelpMessage(), 'Enable or disable Flutter for web.');
-    });
-
-    testWithoutContext('Flutter macOS desktop help string', () {
-      expect(
-        flutterMacOSDesktopFeature.generateHelpMessage(),
-        'Enable or disable support for desktop on macOS.',
+    testWithoutContext('reads from configuration if avaialble', () {
+      const Feature exampleFeature = Feature(
+        name: 'example',
+        master: FeatureChannelSetting(available: true),
       );
-    });
 
-    testWithoutContext('Flutter Linux desktop help string', () {
-      expect(
-        flutterLinuxDesktopFeature.generateHelpMessage(),
-        'Enable or disable support for desktop on Linux.',
+      final FlutterFeatureFlags flags = FlutterFeatureFlags(
+        flutterVersion: FakeFlutterVersion(),
+        featuresConfig: _FakeFeaturesConfig()..cannedResponse[exampleFeature] = true,
+        platform: FakePlatform(),
       );
+      expect(flags.isEnabled(exampleFeature), true);
     });
 
-    testWithoutContext('Flutter Windows desktop help string', () {
-      expect(
-        flutterWindowsDesktopFeature.generateHelpMessage(),
-        'Enable or disable support for desktop on Windows.',
-      );
-    });
+    testWithoutContext('returns false if not available', () {
+      const Feature exampleFeature = Feature(name: 'example');
 
-    testWithoutContext('help string on multiple channels', () {
-      const Feature testWithoutContextFeature = Feature(
-        name: 'example',
-        master: FeatureChannelSetting(available: true),
-        beta: FeatureChannelSetting(available: true),
-        stable: FeatureChannelSetting(available: true),
-        configSetting: 'foo',
+      final FlutterFeatureFlags flags = FlutterFeatureFlags(
+        flutterVersion: FakeFlutterVersion(),
+        featuresConfig: _FakeFeaturesConfig()..cannedResponse[exampleFeature] = true,
+        platform: FakePlatform(),
       );
-
-      expect(testWithoutContextFeature.generateHelpMessage(), 'Enable or disable example.');
+      expect(flags.isEnabled(exampleFeature), false);
     });
 
-    /// Flutter Web
+    FileSystem createFsWithPubspec(String pubspec) {
```

Comment:
Nit: Since we're only using this in one place, can we instead hardcode the contents of `pubspec` here and not take a parameter? That way it's obvious what the contents of the pubspec are without scrolling to the bottom of the test.

Reply:
Done.